PWGUD/Tasks/diffMCDataScanner.cxx

```diff
@@ -0,0 +1,486 @@
+        registry.get<TH2>(HIST("FwdType"))->Fill(1., 0., 1.);
+      }
+      for (auto ii = 0; ii < nTypes; ii++) {
+        if ((fwdtrack.trackType() & (1 << ii)) > 0) {
+          registry.get<TH2>(HIST("FwdType"))->Fill((Double_t)(ii + 2), 0., 1.);
+        }
+      }
+    }
+  }
+};
+
+// MCTruth tracks
+struct MCTracks {
+
+  using CCs = soa::Join<aod::Collisions, aod::McCollisionLabels>;
+  using CC = CCs::iterator;
+
+  void process(CCs const& collisions, aod::McCollisions& McCols, aod::McParticles& McParts)
+  {
+
+    for (auto collision : collisions) {
+
+      // get McCollision which belongs to collision
+      auto MCCol = collision.mcCollision();
+
+      LOGF(info, "Collision %i MC collision %i",
+           collision.globalIndex(), MCCol.globalIndex());
+
+      // get MCParticles which belong to MCCol
+      auto MCPartSlice = McParts.sliceBy(aod::mcparticle::mcCollisionId, MCCol.globalIndex());
+      LOGF(info, "  Number of McParticles %i", MCPartSlice.size());
+
+      // loop over particles
+      float etot = 0.0;
+      float px = 0.0, py = 0.0, pz = 0.0;
+      bool hasDiff = false;
+      int prongs = 0;
+
+      for (auto mcpart : MCPartSlice) {
+        LOGF(info, " MCPart: %i %i %i %i %i - %i %i %i", mcpart.mcCollisionId(), mcpart.isPhysicalPrimary(), mcpart.getProcess(), mcpart.getGenStatusCode(), mcpart.globalIndex(), mcpart.pdgCode(), mcpart.mother0Id(), mcpart.mother1Id());
+        if (mcpart.pdgCode() == 9900110) {
+          LOGF(info, "  rho_diff0 energy: %f", mcpart.e());
+          hasDiff = true;
+        }
+
+        if (hasDiff) {
+          if (mcpart.isPhysicalPrimary() &&
+              (mcpart.getGenStatusCode() == 1 || mcpart.getGenStatusCode() == 2) &&
+              mcpart.mother0Id() != mcpart.mother1Id() &&
+              mcpart.mother1Id() > 0) {
+
+            prongs++;
+            etot += mcpart.e();
+            px += mcpart.px();
+            py += mcpart.py();
+            pz += mcpart.pz();
+          }
+        }
+      }
+      if (hasDiff) {
+        auto mass = TMath::Sqrt(etot * etot - (px * px + py * py + pz * pz));
+        LOGF(info, "  mass of X: %f, prongs: %i", mass, prongs);
+      }
+    }
+  }
+};
+
+// TPC nSigma
+struct TPCnSigma {
+  Produces<aod::nSigmas> nSigmas;
+
+  using TCs = soa::Join<aod::Tracks, aod::TrackSelection, aod::McTrackLabels>;
+  using TCwPIDs = soa::Join<TCs, aod::pidTPCEl, aod::pidTPCMu, aod::pidTPCPi, aod::pidTPCKa, aod::pidTPCPr>;
+
+  void process(TCwPIDs& tracks, aod::McParticles const& mcParticles)
+  {
+    for (auto track : tracks) {
+      if (track.isGlobalTrack()) {
+        nSigmas(track.mcParticle().pdgCode(), track.mcParticle().pt(),
+                track.tpcNSigmaEl(), track.tpcNSigmaMu(), track.tpcNSigmaPi(),
+                track.tpcNSigmaKa(), track.tpcNSigmaPr());
+      }
+    }
+  }
+};
+
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  if (cfgc.options().get<int>("runCase") == 0) {
```

Comment:
Note that these options still not work on hyperloop (just in case)

Reply:
Thanks for the warniong. I am currently using this task mainly for local tests.